cf-java-logging-support-logback/src/main/java/com/sap/hcp/cf/logback/converter/CustomFieldsConverter.java

```diff
@@ -0,0 +1,60 @@
+package com.sap.hcp.cf.logback.converter;
+
+import java.util.Map;
+
+import com.sap.hcp.cf.logging.common.LogContext;
+import com.sap.hcp.cf.logging.common.converter.DefaultCustomFieldsConverter;
+import com.sap.hcp.cf.logging.common.customfields.CustomField;
+
+import ch.qos.logback.classic.pattern.ClassicConverter;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+
+/**
+ * This is a simple {@link ClassicConverter} implementation that converts
+ * key/value pairs stored in {@link CustomField} instances which have been
+ * passed as arguments.
+ * <p>
+ * We allow to types of addition to a log message, either <i>embedded</i>, i.e.
+ * the key/value pairs appear as a list of JSON fields in the message, or as a
+ * nested object where the field name has been specified as an option to this
+ * converter.
+ */
+
+public class CustomFieldsConverter extends ClassicConverter {
+
+	public static final String WORD = "args";
+
+	private DefaultCustomFieldsConverter converter = new DefaultCustomFieldsConverter();
+	private CustomFieldsAdapter customFieldsAdapter = new CustomFieldsAdapter();
+	
+
+	void setConverter(DefaultCustomFieldsConverter converter) {
+		this.converter = converter;
+	}
+
+	void setCustomFieldsAdapter(CustomFieldsAdapter customFieldsAdapter) {
+		this.customFieldsAdapter = customFieldsAdapter;
```

Comment:
The method is for testing purposes only. There it is ensured, that it is called before `start()`. As discussed, we will not ensure this order of calls in the implementation by additional checks, since the method is package private anyways.